src/resource-graph/azext_resourcegraph/custom.py

```diff
@@ -14,90 +14,95 @@
-
-    subs_list = subscriptions or _get_cached_subscriptions()
-
-    if len(subs_list) > __SUBSCRIPTION_LIMIT:
-        subs_list = subs_list[:__SUBSCRIPTION_LIMIT]
-        warning_message = "The query included more subscriptions than allowed. "\
-                          "Only the first {0} subscriptions were included for the results. "\
-                          "To use more than {0} subscriptions, "\
-                          "see the docs for examples: https://aka.ms/arg-error-toomanysubs".format(__SUBSCRIPTION_LIMIT)
+    mgs_list = management_groups
+    if mgs_list is not None and len(mgs_list) > __MANAGEMENT_GROUP_LIMIT:
+        mgs_list = mgs_list[:__MANAGEMENT_GROUP_LIMIT]
+        warning_message = "The query included more management groups than allowed. "\
+                          "Only the first {0} management groups were included for the results. "\
+                          "To use more than {0} management groups, "\
+                          "see the docs for examples: "\
+                          "https://aka.ms/arg-error-toomanysubs".format(__MANAGEMENT_GROUP_LIMIT)
         __logger.warning(warning_message)
 
-    results = []
-    skip_token = None
-    full_query = graph_query
-
-    if include == IncludeOptionsEnum.display_names:
-        try:
-            full_query = _get_extension() + "| " + graph_query
-
-        except Exception as e:
-            __logger.warning("Failed to include displayNames to result. Error: %s", e)
+    subs_list = None
+    if mgs_list is None:
+        subs_list = subscriptions or _get_cached_subscriptions()
+        if subs_list is not None and len(subs_list) > __SUBSCRIPTION_LIMIT:
+            subs_list = subs_list[:__SUBSCRIPTION_LIMIT]
+            warning_message = "The query included more subscriptions than allowed. "\
+                              "Only the first {0} subscriptions were included for the results. "\
+                              "To use more than {0} subscriptions, "\
+                              "see the docs for examples: "\
+                              "https://aka.ms/arg-error-toomanysubs".format(__SUBSCRIPTION_LIMIT)
+            __logger.warning(warning_message)
 
+    results = []
     try:
         result_truncated = False
-        while True:
-            request_options = QueryRequestOptions(
-                top=min(first - len(results), __ROWS_PER_PAGE),
-                skip=skip + len(results),
-                skip_token=skip_token,
-                result_format=ResultFormat.object_array
-            )
-
-            request = QueryRequest(query=full_query, subscriptions=subs_list, options=request_options)
-            response = client.resources(request)  # type: QueryResponse
-            if response.result_truncated == ResultTruncated.true:
-                result_truncated = True
 
+        if first is not None:
+            first = min(first, __ROWS_PER_PAGE)
+        elif skip_token is None:
+            first = __FIRST
```

Comment:
Why assign first when skip_token is None

Reply:
Yes, it's page size. It's actually renamed to 'top' on a server side but I didn't want to introduce this breraking change.

This is the case where is user didn't pass neither skip token nor first explicitly, we're setting first to a default value from help message which is 100.